modules/hub/src/ChainsawService.ts

```diff
@@ -268,14 +268,70 @@ export default class ChainsawService {
   private async processDidStartExitChannel(chainsawId: number, event: DidStartExitChannelEvent) {
     const onchainChannel = await this.signerService.getChannelDetails(event.user)
 
-    const disputeRecord = await this.channelDisputesDao.getActive(event.user)
+    let disputeRecord = await this.channelDisputesDao.getActive(event.user)
     if (!disputeRecord) {
       // dispute might not have been initiated by us, so we need to add it here
-      await this.channelDisputesDao.create(event.user, 'Dispute caught by chainsaw', chainsawId, null, onchainChannel.channelClosingTime)
+      disputeRecord = await this.channelDisputesDao.create(event.user, 'Dispute caught by chainsaw', chainsawId, null, onchainChannel.channelClosingTime)
+    } else {
+      await this.channelDisputesDao.setExitEvent(disputeRecord.id, chainsawId, onchainChannel.channelClosingTime)
+    }
+
+    // check if sender was user
+    if (event.senderIdx == 0) {
+      LOG.info(`Hub inititated the challenge, so no need to respond; event ${prettySafeJson(event)}`)
       return
     }
 
-    await this.channelDisputesDao.setExitEvent(disputeRecord.id, chainsawId, onchainChannel.channelClosingTime)
+    let txn
+    let data
+    const latestUpdate = await this.channelsDao.getLatestExitableState(event.user)
+    if (event.txCountGlobal <= latestUpdate.state.txCountGlobal) {
+      LOG.info(`Channel has not exited with the latest state, hub will respond! event ${prettySafeJson(event)}`)
```

Comment:
Should probs also have a log message that we're emptying channel, along with the event and the latest channel event.